src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -523,4 +523,32 @@ public function getChildNodeDefinitions()
     {
         return $this->children;
     }
+
+    /**
+     * Finds a node defined by the given $nodePath.
+     *
+     * @param string $nodePath The path of the node to find. e.g "doctrine.orm.mappings"
+     *
+     * @return NodeDefinition
+     */
+    public function find(string $nodePath): NodeDefinition
+    {
+        if ($nodePath === $this->name) {
+            return $this;
+        }
+
+        $firstPathSegment = strtok($nodePath, $this->pathSeparator);
```

Comment:
`$pathSeparator` doesn't necessarily have to be a single character if I don't miss anything. Using more than one character as the separator would break here.

Reply:
Actually this shouldn't be a problem: https://3v4l.org/ET6V6

Reply:
But something like https://3v4l.org/Aucsi should not happen, right?

Reply:
Yes you are right. Using a separator like `.|` with a given node name like `mailer.config` would break the find() method. I added a test for this and removed the `strtok` method in favour of `strpos` and `substr` See: 9e7daa3